### PR TITLE
add unit test for 'validate' package

### PR DESCRIPTION
**Pull request by carmark**

### CODE REVIEW

**Comment by carmark** [comment]
ok to test

File: validate/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		if mi != c.minor {
+			t.Errorf("Unexpected minor, should %v, but got %v", c.minor, mi)
+		}
+	}
+}
+
+func TestValidateKernelVersion(t *testing.T) {
+	const (
+		standardDesc = "Versions >= 2.6 are supported. 3.0+ are recommended"
+		errorDesc    = "Could not parse kernel version"
```
Comment (timstclair) [inline validate/validate_test.go]: I'd prefer not to duplicate these messages. Move these constants to validate.go, and use in both files.

File: validate/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		{"3.1", Unknown, errorDesc},
+	}
+
+	for _, c := range cases {
+		res, desc := validateKernelVersion(c.version)
+		if res != c.result {
+			t.Errorf("Unexpected result, should %v, but got %v", c.result, res)
+		}
+		if !strings.Contains(desc, c.desc) {
+			t.Errorf("Unexpected desction, should %v, but got %v", c.desc, desc)
```
Comment (timstclair) [inline validate/validate_test.go]: s/desction/description/

File: validate/validate_test.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		}
+	}
+}
+
+func TestValidateDockerVersion(t *testing.T) {
+	const (
+		standardDesc = "Versions >= 1.0 are supported. 1.2+ are recommended"
+		errorDesc    = "Could not parse docker version"
```
Comment (timstclair) [inline validate/validate_test.go]: Ditto re: duplicating messages

File: validate/validate_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		{"-1.-1beta", -1, -1, nil},
+		{"0.1", -1, -1, fmt.Errorf("have error")},
+		{"0", -1, -1, fmt.Errorf("have error")},
+		{"beta", -1, -1, fmt.Errorf("have error")},
+	}
+
+	for _, c := range cases {
+		ma, mi, e := getMajorMinor(c.version)
+		if (e != nil) != (c.err != nil) {
+			t.Errorf("Unexpected err, should %v, but got %v", c.err, e)
```
Comment (timstclair) [inline validate/validate_test.go]: I like to put the test case number in all error messages, e.g. `"[%d] ....", i`, where `i` is the first value returned by `range`

**Comment by timstclair** [comment]
Thanks!

**Comment by carmark** [comment]
@timstclair 
Thanks for your review, I made the error message more useful by adding test case ID and fixed spelling mistakes.  I kept the `standardDesc` and `errorDesc` in validate/validate_test.go, since they are only used in test scenario.

**Comment by timstclair** [comment]
LGTM. Do you have merge permissions?

**Comment by carmark** [comment]
@timstclair 
No, I have no merge permission, could you please merge this?

Thanks,

**Pull request merged**